Automattic/wp-calypso
# Update plan-specific flows to use generic plans step

This is a temporary fix to avoid the BSOD (blue screen of death) when switching to the `ecommerce-onboarding` flow after selecting the `Online store` site type. It swaps the more specific plans step types for the generic `plans` step.

This makes the onboarding flow slightly longer for users who pre-select a plan from wordpress.com (they need to select the plan again from the `plans` step), however it prevents the BSOD issue of selecting the `Online store` site type and then getting a blank screen.

The underlying issue is tracked in https://github.com/Automattic/wp-calypso/issues/32506 and a potential _proper_ fix is being explored in https://github.com/Automattic/wp-calypso/pull/34885

Once an underlying fix is merged, we can then return to using plans-specific steps for each of these flows.

Screenshot of the BSOD issue:

![Aug-01-2019 18-01-23](https://user-images.githubusercontent.com/14988353/62277649-ee1e6780-b489-11e9-8c99-708ce533faf7.gif)

#### Changes proposed in this Pull Request

* Update the `free`, `personal`, `premium`, and `business` flows to use the generic `plans` step so that the flow is compatible with the `ecommerce-onboarding` flow when a user switches flow after selecting an `Online store` site type.
* Update e2e tests for creating a personal, premium, and business site from the `/create` page of WordPress.com to include the additional plans step.

#### Testing instructions



* From an incognito window, go to http://calypso.localhost:3000/start/personal
* Log in on the user step
* From the Site type step, select `Online store`, and it should take you to the domains step on the `ecommerce-onboarding` flow, and you should see: http://calypso.localhost:3000/start/ecommerce-onboarding/domains in the address bar
* Click back, and you should be on the `site-type` step
* Select any other site type (e.g. blog) and you should be on the `site-topic-with-preview` step (e.g. http://calypso.localhost:3000/start/personal/site-topic-with-preview )

Repeat the above steps for:

* http://calypso.localhost:3000/start/premium
* http://calypso.localhost:3000/start/business
* http://calypso.localhost:3000/start/free

For each flow, ensure that a site can be created, and you arrive on the checklist page after signup.
